spec/controllers/reservations_controller_spec.rb

```diff
@@ -453,58 +404,57 @@
   end
 
   describe '#edit (GET /reservations/:id/edit)' do
-    ## Unhappy paths
+    # SMELLS:
+    #   - message chain
     it_behaves_like 'inaccessible by banned user' do
-      before { get 'edit', id: @reservation.id }
+      before { get 'edit', id: 1 }
     end
 
     context 'when accessed by patron' do
       before(:each) do
-        sign_in @user
-        get 'edit', id: @reservation.id
+        mock_user_sign_in
+        get 'edit', id: 1
       end
-      include_examples 'cannot access page'
+      include_examples 'redirected request'
     end
 
     context 'when accessed by checkout person disallowed by settings' do
       before(:each) do
-        sign_in @checkout_person
-        allow(@ac).to receive(:checkout_persons_can_edit).and_return(false)
-        get 'edit', id: @reservation.id
+        mock_app_config(AC_DEFAULTS.merge(checkout_persons_can_edit: false))
+        mock_user_sign_in(UserMock.new(:checkout_person))
+        get 'edit', id: 1
       end
-      include_examples 'cannot access page'
+      include_examples 'redirected request'
     end
 
-    ## Happy paths
     shared_examples 'can access edit page' do
-      it 'assigns @reservation' do
-        expect(assigns(:reservation)).to eq(@reservation)
+      let!(:item) { EquipmentItemMock.new(id: 1, name: 'Name') }
+      before do
+        model = EquipmentModelMock.new(traits: [[:with_item, item: item]])
+        res = ReservationMock.new(traits: [:findable], equipment_model: model)
+        get :edit, id: res.id
       end
       it 'assigns @option_array as Array' do
         expect(assigns(:option_array)).to be_an Array
       end
       it 'assigns @option_array with the correct contents' do
-        expect(assigns(:option_array)).to\
-          eq @reservation.equipment_model.equipment_items
-                         .collect { |e| [e.name, e.id] }
+        expect(assigns(:option_array)).to eq([[item.name, item.id]])
       end
       it { is_expected.to render_template(:edit) }
     end
 
     context 'when accessed by checkout person allowed by settings' do
+      # FIXME: permissions fail before it can hit controller action
```

Comment:
Yeah, my findings were similar, it doesn't actually hit the :edit controller action, and instead just fails since it thinks checkout people can't edit the reservation